### ISSUE TITLE
[Feature Request] Alternate analog toggle chord

### ISSUE THREAD

**Issue by Sanaki** [libretro/beetle-psx-libretro#699]
Right now, the only chord available for analog toggle is L1+L2+R1+R2+Select+Start. This is also the chord used internally by a significant number of PS1 games to trigger a soft reset. This makes it completely non-viable for a number of games. I'd like to propose a second option under "Enable Dualshock Analog Mode Toggle" to select the preferred button chord. Good options would include L1+L2+L3, R1+R2+R3, or even L1+L2+R3 and R1+R2+L3 for people with less manual dexterity. The retroarch menu toggle list has some other good options as well.

Personally, I'd go with R1+R2+R3 for my own use. It's easy to enter and shouldn't conflict with any game or any currently available chord.

There's also the prospect of an option to enable analog to digital on the left stick automatically when analog mode is toggled off in this fashion, but that may be a topic for another issue.

**Comment by hizzlekizzle** [comment]
I don't think the analog-to-digital thing would be possible, since it would be breaking the frontend/core barrier.

The chording could be changed just with core modifications, but as I recall this is just something that's hardcoded upstream, so it may take some digging to unwind.

**Comment by simias** [comment]
I implemented this very feature in my own core a little while ago so I can share a few insights:

- Regarding the key combo I settled on Select + R3 by default since it's fairly easy to type and remember while still being rather unlikely to be used by games. Alternatively I added an option in the core to change this combo to either "Select + L3" or "L3 + R3" to try and cover all the bases. AFAIK few games  use the R/L3 buttons, let alone as a combo with the hard-to-reach select button.

- IMO this is really just a weakness of libretro: there ought to be "general purpose" buttons in the frontend that could be used for core-specific functions, such as additional controller buttons. This way the user could remap the analog toggle to whatever they want in the frontend. It wouldn't be too hard to add new buttons in the libretro interface, the only breaking change would be to the experimental `RETRO_ENVIRONMENT_GET_INPUT_BITMASKS` API which returns the state of the input in a (signed!) 16bit integer. The fix would simply be to have this interface return an uint32_t to allow for 32buttons, for instance. I really think it would be a worthwhile change to make but I can't be bothered to do it myself...

- It might actually be possible to automatically detect if a game supports analog input without requiring user intervention, by seeing if it allows the full analog sequence to terminate instead of bailing out halfway through. I haven't actually tested that theory though, so maybe it wouldn't work as well as I imagine.

Regarding the auto analog-to-digital it might be fine as an option, although I personally find that it's so clunky to use in most games that I can't imagine ever wanting that turned on automatically.
